### PR TITLE
Use unlock pattern everywhere

### PR DESCRIPTION
**Pull request by RoiEXLab**
I had some fun with structural replace

### CODE REVIEW

File: game-app/game-core/src/main/java/games/strategy/engine/data/ResourceCollection.java [hunk 1]
```diff
@@ -172,21 +169,8 @@ private static String toString(
     }
     final StringBuilder sb = new StringBuilder();
     Resource pus = null;
-    data.acquireReadLock();
-    try {
+    try (GameData.Unlocker ignored = data.acquireReadLock()) {
       pus = data.getResourceList().getResource(Constants.PUS);
-    } catch (final NullPointerException e) {
```
Comment (asvitkine) [inline game-app/game-core/src/main/java/games/strategy/engine/data/ResourceCollection.java:178]: This removal seems unrelated to the cleanup. Can you restore?

File: game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/ConcurrentBattleCalculator.java [hunk 2]
```diff
@@ -124,19 +124,14 @@ private void createWorkers(final GameData data) {
           Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory();
       final Version engineVersion = Injections.getInstance().getEngineVersion();
       final byte[] serializedData;
-      try {
-        // Serialize the data, then release lock on it so game can continue (ie: we don't want to
```
Comment (asvitkine) [inline game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/ConcurrentBattleCalculator.java:128]: Can you restore this comment?

**Review by asvitkine — APPROVED** [review]
Nice! Just a couple places where I think some things were unintentionally removed.

**Comment by RoiEXLab** [comment]
@asvitkine The accidental removal thing has been fixed

File: game-app/game-core/src/main/java/games/strategy/engine/chat/PlayerChatRenderer.java [hunk 3]
```diff
@@ -89,11 +90,8 @@ private String getNodeLabelWithPlayers(final UserName userName) {
   }
 
   private static PlayerList getPlayerList(final IGame game) {
-    game.getData().acquireReadLock();
-    try {
+    try (GameData.Unlocker ignored = game.getData().acquireReadLock()) {
```
[the remark this answers is not in the document]
Comment (asvitkine) [reply game-app/game-core/src/main/java/games/strategy/engine/chat/PlayerChatRenderer.java:93]: Yes, I did notice that as well when I was applying this pattern. It still fixes the specific error of trying to unlock an object that wasn't locked, but the inner code is susceptible to the data its using being swapped underneath it.

I think if we eventually go with your idea that the Unlocker is actually an Accessor and all getters on the gameData are done through that object, that can solve it. But that's a much bigger change.
